src/main/java/pages/mainPage/MainPageProduct.java

```diff
@@ -0,0 +1,51 @@
+package pages.mainPage;
+
+import lombok.Getter;
+import models.ProductModel;
+import org.openqa.selenium.By;
+import pages.BasePage;
+
+import java.math.BigDecimal;
+
+@Getter
+public class MainPageProduct {
+    protected BasePage basePage;
+    private final String productXpath;
+
+    public MainPageProduct(int productIndex, String sectionXpath) {
+        this.basePage = new BasePage();
+        this.productXpath = sectionXpath + String.format("//div[@class='thumbnails list-inline']/div[%s]", productIndex);
+        System.out.println(productXpath);
+    }
+
+    public ProductModel getProductsDetails() {
+        return ProductModel.builder()
+                .name(getProductName())
+                .price(getPrice())
+                .build();
+    }
+
+    public void clickAddToCartButton() {
+        By outOfStockElement = By.xpath(productXpath + "//span[@class='nostock']");
+        By addToCartButton = By.xpath(productXpath + "//a[@class='productcart']");
+
+        if (basePage.isElementDisplayed(outOfStockElement)) {
+            System.out.println("Product is out of stock and cannot be added to the cart.");
+        } else if (basePage.isElementDisplayed(addToCartButton)) {
+            basePage.clickOnElement(addToCartButton);
+        } else {
+            System.out.println("Add to cart button is not present for the product.");
```

Comment:
Instead of displaying information on the console with ``System.out.println()`` you might want to abort the method by throwing an exception. The method is called ``clickAddToCartButton()`` so if the click fails, the test will be aborted (thanks to throwing an exception) with the corresponding information on the console as well. 
Leaving ``System.out.println()`` will not abort the test, which can possible add a false positive to the test.
```java
    if (basePage.isElementDisplayed(outOfStockElement)) {
        throw new IllegalStateException("Product is out of stock and cannot be added to the cart.");
    } else if (basePage.isElementDisplayed(addToCartButton)) {
        basePage.clickOnElement(addToCartButton);
    } else {
        throw new NoSuchElementException("Add to cart button is not present for the product.");
    }
```